File: get-started/go/getuser.go
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

package main

import (
	"context"
	"fmt"

	"getuser/client"

	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
	azure "github.com/microsoft/kiota-authentication-azure-go"
	http "github.com/microsoft/kiota-http-go"
)

func main() {
	clientId := "YOUR_CLIENT_ID"

	// The auth provider will only authorize requests to
	// the allowed hosts, in this case Microsoft Graph
	allowedHosts := []string{"graph.microsoft.com"}
	graphScopes := []string{"User.Read"}

	credential, err := azidentity.NewDeviceCodeCredential(&azidentity.DeviceCodeCredentialOptions{
		ClientID: clientId,
		UserPrompt: func(ctx context.Context, dcm azidentity.DeviceCodeMessage) error {
			fmt.Println(dcm.Message)
			return nil
		},
	})

	if err != nil {
		fmt.Printf("Error creating credential: %v\n", err)
	}

	authProvider, err := azure.NewAzureIdentityAuthenticationProviderWithScopesAndValidHosts(
		credential, graphScopes, allowedHosts)

	if err != nil {
		fmt.Printf("Error creating auth provider: %v\n", err)
	}

	adapter, err := http.NewNetHttpRequestAdapter(authProvider)

	if err != nil {
		fmt.Printf("Error creating request adapter: %v\n", err)
	}

	client := client.NewGraphApiClient(adapter)

	me, err := client.Me().Get()

	if err != nil {
		fmt.Printf("Error getting user: %v\n", err)
	}

	fmt.Printf("Hello %s, your ID is %s\n", *me.GetDisplayName(), *me.GetId())
}

File: get-started/ruby/get_user.rb
# frozen_string_literal: true
require 'microsoft_kiota_serialization'
require 'microsoft_kiota_abstractions'
require 'microsoft_kiota_authentication_oauth'
require 'microsoft_kiota_nethttplibrary'
require_relative './client/api_client'

tenant_id = 'TENANT_ID'
client_id = 'CLIENT_ID'
client_secret = 'CLIENT_SECRET'
auth_code = 'AUTH_CODE'
redirect_uri = 'REDIRECT_URI'

# The auth provider will only authorize requests to
# the allowed hosts, in this case Microsoft Graph
allowed_hosts = ['graph.microsoft.com']
graph_scopes = ['User.Read']
token_request_context = MicrosoftKiotaAuthenticationOAuth::AuthorizationCodeContext.new(tenant_id, client_id, client_secret, redirect_uri, auth_code)

auth_provider = MicrosoftKiotaAuthenticationOAuth::OAuthAuthenticationProvider.new(token_request_context, allowed_hosts, graph_scopes)

request_adapter = MicrosoftKiotaNethttplibrary::NetHttpRequestAdapter.new(auth_provider, MicrosoftKiotaSerialization::JsonParseNodeFactory.new, MicrosoftKiotaSerialization::JsonSerializationWriterFactory.new)

client = Graph::ApiClient.new(request_adapter)

me = client.me.get

puts "Hi! My name is #{me.display_name}, and my ID is #{me.id}."
